spec/mailers/teacher_mailer_spec.rb

```diff
@@ -417,4 +417,56 @@
 
     it_behaves_like "an observable mailer", "references_requested"
   end
+
+  describe "#references_reminder" do
+    let!(:reference_request) do
+      create(
+        :reference_request,
+        requested_at: Date.new(2020, 1, 1),
+        assessment: application_form.assessment,
+      )
+    end
+    subject(:mail) do
+      described_class.with(
+        teacher:,
+        reference_requests: [reference_request],
+      ).references_reminder
+    end
+
+    describe "#subject" do
+      subject(:subject) { mail.subject }
+
+      context "with no reminder emails" do
+        it { is_expected.to eq("Waiting on references – QTS application") }
+      end
+
+      context "with one reminder email" do
+        it do
+          teacher.reminder_emails.create!
+          is_expected.to eq(
+            "Your references only have two weeks left to respond",
+          )
+        end
+      end
+    end
+
+    describe "#to" do
+      subject(:to) { mail.to }
+
+      it { is_expected.to eq(["teacher@example.com"]) }
+    end
+
+    describe "#body" do
+      subject(:body) { mail.body.encoded }
+
+      it { is_expected.to include("Dear First Last") }
+      it do
+        is_expected.to include(
+          "We’re still waiting for a response from one or more of the references you provided to verify your work history."
+        )
+      end
```

Comment:
Could we also add some tests to check the content related to the individual reference requests also comes through correctly?

Reply:
I'm thinking something that would have caught https://github.com/DFE-Digital/apply-for-qualified-teacher-status/pull/1778#discussion_r1389159635